src/runtime/d3d12compute.cpp

```diff
@@ -667,14 +679,28 @@ void release_d3d12_object<d3d12_compute_pipeline_state>(d3d12_compute_pipeline_s
 
 extern WEAK halide_device_interface_t d3d12compute_device_interface;
 
+static d3d12_buffer *peel_buffer(struct halide_buffer_t *hbuffer) {
+    TRACELOG;
+    halide_assert(user_context, (hbuffer != NULL));
+    halide_assert(user_context, (hbuffer->device_interface == &d3d12compute_device_interface));
+    d3d12_buffer *dbuffer = reinterpret_cast<d3d12_buffer*>(hbuffer->device);
+    halide_assert(user_context, (dbuffer != NULL));
+    halide_assert(user_context, (dbuffer->halide));
+    return dbuffer;
+}
+
+static const d3d12_buffer *peel_buffer(const struct halide_buffer_t *hbuffer) {
+    return peel_buffer( const_cast<halide_buffer_t*>(hbuffer) );
+}
+
 WEAK int wrap_buffer(struct halide_buffer_t *hbuffer, d3d12_buffer *dbuffer) {
     halide_assert(user_context, (hbuffer->device == 0));
     if (hbuffer->device != 0) {
         return -2;
     }
 
     halide_assert(user_context, (dbuffer->resource != NULL));
-    halide_assert(user_context, (dbuffer->halide == NULL));
+    halide_assert(user_context, (!dbuffer->halide));
```

Comment:
I think these two lines should be consistently styled; I prefer the more explicity `== NULL` but not sure what is appropriate Halide coding style.